Source/bmalloc/libpas/src/libpas/pas_report_crash.c

```diff
@@ -67,6 +67,13 @@ static PAS_ALWAYS_INLINE kern_return_t PAS_WARN_UNUSED_RETURN pas_update_report_
     return KERN_SUCCESS;
 }
 
+// Function to be called to check if PGM has been enabled on this process at any point,
+// regardless of if it has since been disabled, or if any guarded allocations were made.
+kern_return_t pas_pgm_enabled_on_process(void)
+{
+    return pas_probabilistic_guard_malloc_has_been_used ? KERN_SUCCESS : KERN_FAILURE;
+}
+
```

Comment:
You should define `pas_probabilistic_guard_malloc_enabled` function in `pas_probabilistic_guard_malloc_allocator.h` and `pas_probabilistic_guard_malloc_allocator.c` and use it.